gymnasium/spaces/utils.py

```diff
@@ -271,7 +271,13 @@ def _unflatten_box_multibinary(
 
 @unflatten.register(Discrete)
 def _unflatten_discrete(space: Discrete, x: NDArray[np.int64]) -> np.int64:
-    return space.start + np.nonzero(x)[0][0]
+    nonzero = np.nonzero(x)
+    if len(nonzero[0]) == 0:
+        raise ValueError(
+            f"{x} is not a valid one-hot encoded vector and can not be unflattened to space {space}. "
+            f"Not all valid values in a flattened space can be unflattened."
```

Comment:
Change from `values` to `samples`, "not all samples from a flattened space can be unflattened"

Reply:
Done.